documentation/reference/ci_cd.md

```diff
@@ -0,0 +1,491 @@
+| `api`              | 🛠             |           | 🛠                  | 🚀    | 🚀          |            |
+| `frontend`         |               |           |                    |       |             | 🚀         |
+
+🚀 implies that the image is published to GHCR. 🛠️ implies that the image is
+built but not published.
+
+```{note}
+The `upstream_db` image is only built here and reused in other workflows for
+convenience and speed, it is never published.
+```
+
+**Outputs:**
+
+```typescript
+interface Output {
+  do_build: "true" | "false" // whether one or more images are to be built
+  build_matrix: {
+    image: (
+      | "upstream_db"
+      | "catalog"
+      | "ingestion_server"
+      | "api"
+      | "api_nginx"
+      | "frontend"
+    )[] // the names of the image to be built
+    include: (
+      | { image: "upstream_db"; context: "docker/upstream_db"; target: "db" }
+      | { image: "catalog"; context: "catalog"; target: "cat" }
+      | {
+          image: "ingestion_server"
+          context: "ingestion_server"
+          target: "ing"
+        }
+      | { image: "api"; context: "api"; target: "api" }
+      | { image: "api_nginx"; context: "api"; target: "nginx" }
+    )[] // additional information about images to be built
+  }
+  do_publish: "true" | "false" // whether one or more images are to be published
+  publish_matrix: {
+    image: ("catalog" | "ingestion_server" | "api" | "api_nginx" | "frontend")[] // the name of the image to be published
+  }
+}
+```
+
+## Universal jobs
+
+### `build-images`
+
+Builds images as dictated by the [`determine-images`](#determine-images) job.
+Since the images are built in a matrix, some image-specific steps are
+conditionally run using an `if` expression. Since any unused Docker build
+arguments have no effect, the job simply populates all build arguments needed by
+all images.
+
+Images built by this job are published as `.tar` artifacts and can be loaded
+into other jobs that need them using the [`load-img`](#load-img) action.
+
+This job is only run if there is at least one image needed to build, based on
+the `do_build` output of the [`determine-images`](#determine-images) job.
+
+## Catalog jobs
+
+### `test-cat`
+
+Runs tests for the catalog using the `catalog/test` recipe. Tests are run inside
+a Docker container so neither Python nor Node.js needs to be installed.
+
+This job is skipped if the catalog codebase has not changed. Its counterparts
+are [`test-ing`](#test-ing) for the ingestion server, [`test-api`](#test-api)
+for the API and [`nuxt-build`](#nuxt-build) for the frontend.
+
+```{note}
+This job is treated as the proof of functionality for publishing Docker images
+for the catalog.
+```
+
+### `catalog-checks`
+
+Runs tests for the catalog using the `catalog/generate-dag-docs` recipe. The job
+generates a new file from a Docker container so Python does not need to be
+installed. However, the job runs linting on the new file, so Node.js needs to be
+set up.
```

Comment:
The middle sentence is a little unclear unless you already know what this job does. I had to re-read it a few times to realise what was being conveyed by it. Suggested edit for clarity:

```suggestion
Runs tests for the catalog using the `catalog/generate-dag-docs` recipe. The job runs inside a Docker container so does not require python to be installed. However, the job generates a new documentation page that must be linted, so `setup-env` is required to install linting dependencies.
```

Hopefully that is a correct understanding of this. If not, it still needs further clarification :sweat_smile:

Reply:
Updated in a87e889.